editor/svgutils.js

```diff
@@ -529,19 +529,387 @@ svgedit.utilities.getBBox = function(elem) {
+					['L',[x, y]],
+					['Z',[]]
+				]);
+			} else {
+				d = svgedit.utilities.getPathDFromSegments([
+					['M',[x, y+ry]],
+					['C',[x, y+ry/num, x+rx/num, y, x+rx, y]],
+					['L',[x+w-rx, y]],
+					['C',[x+w-rx/num, y, x+w, y+ry/num, x+w, y+ry]],
+					['L',[x+w, y+h-ry]],
+					['C',[x+w, y+h-ry/num, x+w-rx/num, y+h, x+w-rx, y+h]],
+					['L',[x+rx, y+h]],
+					['C',[x+rx/num, y+h, x, y+h-ry/num, x, y+h-ry]],
+					['L',[x, y+ry]],
+					['Z',[]]
+				]);
+			}
+			break;
+		default:
+			break;
+	}
+
+	return d;
+
+};
+
+svgedit.utilities.addAttributesForConvertToPath = function( elem, attrs) {
+	// TODO: make this list global so that we can properly maintain it
+	// TODO: what about @transform, @clip-rule, @fill-rule, etc?
+	$.each(['marker-start', 'marker-end', 'marker-mid', 'filter', 'clip-path'], function() {
+		if (elem.getAttribute(this)) {
+			attrs[this] = elem.getAttribute(this);
+		}
+	});
+};
+
+// Function: getBBoxOfElementAsPath
+// Get the BBox of an element-as-path
 //
 // Parameters:
-// elem - Optional DOM element to get the angle for
+// elem - The DOM element to be probed
+// addSvgElementFromJson - Function to add the path element to the current layer. See canvas.addSvgElementFromJso
+// pathActions - If a transform exists, pathActions.resetOrientation() is used. See: canvas.pathActions.
+//
+// Returns:
+// The resulting path's bounding box object.
+svgedit.utilities.getBBoxOfElementAsPath = function(elem, addSvgElementFromJson, pathActions) {
+
+	var attrs = {}
+
+	// any attribute on the element not covered by the above
+	svgedit.utilities.addAttributesForConvertToPath( elem, attrs)
+
+	var path = addSvgElementFromJson({
+		'element': 'path',
+		'attr': attrs
+	});
+
+	var eltrans = elem.getAttribute('transform');
+	if (eltrans) {
+		path.setAttribute('transform', eltrans);
+	}
+
+	var id = elem.id;
+	var parent = elem.parentNode;
+	if (elem.nextSibling) {
+		parent.insertBefore(path, elem);
+	} else {
+		parent.appendChild(path);
+	}
+
+	var d = svgedit.utilities.getPathDFromElement( elem);
+	if( d)
+		path.setAttribute('d', d);
+	else
+		path.parentNode.removeChild(path);
+
+	// Get the correct BBox of the new path, then discard it
+	pathActions.resetOrientation(path);
+	var bb = false;
+	try {
+		bb = path.getBBox();
+	} catch(e) {
+		// Firefox fails
+	}
+	path.parentNode.removeChild(path);
+	return bb;
+}
+
+// Function: convertToPath
+// Convert selected element to a path.
+//
+// Parameters:
+// elem - The DOM element to be converted
+// attrs - Apply attributes to new path. see canvas.convertToPath
+// addSvgElementFromJson - Function to add the path element to the current layer. See canvas.addSvgElementFromJso
+// pathActions - If a transform exists, pathActions.resetOrientation() is used. See: canvas.pathActions.
+// clearSelection - see canvas.clearSelection
+// addToSelection - see canvas.addToSelection
+// history - see svgedit.history
+// addCommandToHistory - see canvas.addCommandToHistory
```

Comment:
Same comment about all these svgcanvas-specific functions you are passing into a "utility" function.  It means this isn't really a utility function after all.

I don't want to derail the huge performance wins you've made with this patch, it's great - but sending functions into this function to be called just ultimately makes the code harder to understand later.  Maybe think about it in future patches?